service/history/workflow/cache/cache.go

```diff
@@ -220,7 +222,7 @@ func (c *CacheImpl) lockWorkflowExecution(ctx context.Context,
 	if deadline, ok := ctx.Deadline(); ok {
 		var cancel context.CancelFunc
 		if headers.GetCallerInfo(ctx).CallerType != headers.CallerTypeAPI {
```

Comment:
@pdoerner Can you send out a separate PR for your operator callerType change to exclude operator from here as well?